src/XacroParser.js

```diff
@@ -21,6 +21,8 @@ export class XacroParser {
         this.requirePrefix = true;
         this.localProperties = true;
         this.rospackCommands = {};
+        this.arguments = {};
+        this.argumentDefaults = {};
```

Comment:
I think `argumentDefaults` can be defined as a local variable down with the rest of them [here](https://github.com/gkjohnson/xacro-parser/blob/47f17ecb9e82d68a3d69187fdef49989e2701add/src/XacroParser.js#L528-L533). If this is on the scope of parser then it can cause conflicts if the same parser instance is reused. Sorry I was unclear!

Reply:
I see, didn't think about reusing the same parser. I've moved it to a local variable — do you want me to also thread it through the arguments of all the helper functions? It doesn't seem necessary in order to make things work.

Reply:
> do you want me to also thread it through the arguments of all the helper functions

No that should be fine -- the reason other objects are explicitly passed is to ensure the right scope is respected when parsing. These default args are always global in scope, correct?

Reply:
Yep: http://wiki.ros.org/xacro#Rospack_commands

> In xacro files, `args` is a _global_ dictionary accessible from within any included file or macro.

I also added a test for this with an include: https://github.com/gkjohnson/xacro-parser/pull/79/files#diff-104385f6f5dde804971dd2a206e42bbf2d2ee1fcf28c285d44ac71bfe33aacaeR986